src/query/storage/promremote/storage_test.go

```diff
@@ -0,0 +1,253 @@
+		assert.NotNil(t, promShortRetention.GetLastRequest())
+		assert.Nil(t, promMediumRetention.GetLastRequest())
+		assert.Nil(t, promLongRetention.GetLastRequest())
+	})
+
+	t.Run("send medium retention write", func(t *testing.T) {
+		reset()
+		err := sendWrite(storagemetadata.Attributes{
+			Resolution: 5 * time.Minute,
+			Retention:  720 * time.Hour,
+		})
+		require.NoError(t, err)
+		assert.Nil(t, promShortRetention.GetLastRequest())
+		assert.NotNil(t, promMediumRetention.GetLastRequest())
+		assert.Nil(t, promLongRetention.GetLastRequest())
+	})
+
+	t.Run("send long retention write", func(t *testing.T) {
+		reset()
+		err := sendWrite(storagemetadata.Attributes{
+			MetricsType: storagemetadata.AggregatedMetricsType,
+			// Should be ignored when type is unagg
+			Resolution: 10 * time.Minute,
+			Retention:  8760 * time.Hour,
+		})
+		require.NoError(t, err)
+		assert.Nil(t, promShortRetention.GetLastRequest())
+		assert.Nil(t, promMediumRetention.GetLastRequest())
+		assert.NotNil(t, promLongRetention.GetLastRequest())
+	})
+
+	t.Run("send write to multiple instances configured with same retention", func(t *testing.T) {
+		reset()
+		err := sendWrite(storagemetadata.Attributes{
+			MetricsType: storagemetadata.AggregatedMetricsType,
+			// Should be ignored when type is unagg
+			Resolution: 10 * time.Minute,
+			Retention:  8760 * time.Hour,
+		})
+		require.NoError(t, err)
+		assert.Nil(t, promShortRetention.GetLastRequest())
+		assert.Nil(t, promMediumRetention.GetLastRequest())
+		assert.NotNil(t, promLongRetention.GetLastRequest())
+		assert.NotNil(t, promLongRetention2.GetLastRequest())
+	})
+
+	t.Run("send unconfigured retention write", func(t *testing.T) {
+		reset()
+		err := sendWrite(storagemetadata.Attributes{
+			MetricsType: storagemetadata.AggregatedMetricsType,
+			// Should be ignored when type is unagg
+			Resolution: 5*time.Minute + 1,
+			Retention:  720 * time.Hour,
+		})
+		require.NoError(t, err)
+		err = sendWrite(storagemetadata.Attributes{
+			MetricsType: storagemetadata.AggregatedMetricsType,
+			// Should be ignored when type is unagg
+			Resolution: 5 * time.Minute,
+			Retention:  720*time.Hour + 1,
+		})
+		require.NoError(t, err)
+		assert.Nil(t, promShortRetention.GetLastRequest())
+		assert.Nil(t, promMediumRetention.GetLastRequest())
+		assert.Nil(t, promLongRetention.GetLastRequest())
+	})
+
+	t.Run("error should not prevent sending to other instances", func(t *testing.T) {
+		reset()
+		promLongRetention.SetError(errors.New("test err"))
+		err := sendWrite(storagemetadata.Attributes{
+			MetricsType: storagemetadata.AggregatedMetricsType,
+			// Should be ignored when type is unagg
+			Resolution: 10 * time.Minute,
+			Retention:  8760 * time.Hour,
+		})
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "test err")
+		assert.NotNil(t, promLongRetention2.GetLastRequest())
+	})
+}
+
+func closeStorage(t *testing.T, s storage.Storage) {
+	require.NoError(t, s.Close())
+}
```

Comment:
Nit - you could make it more generic (and perhaps we already have something like this):
```suggestion
func closeWithCheck(t *testing.T, c io.Closer) {
	require.NoError(t, c.Close())
}
```